packages/xforms-engine/src/client/media/MediaResource.ts

```diff
@@ -0,0 +1,17 @@
+import type { AudioMediaResource } from './AudioMediaResource.ts';
+import type { MediaResourceType } from './BaseMediaResource.ts';
+import type { ImageMediaResource } from './ImageMediaResource.ts';
+import type { UnknownMediaResource } from './UnknownMediaResource.ts';
+import type { VideoMediaResource } from './VideoMediaResource.ts';
+
+interface MediaResourceByType {
+	readonly MEDIA_RESOURCE_TYPE_AUDIO: AudioMediaResource;
+	readonly MEDIA_RESOURCE_TYPE_IMAGE: ImageMediaResource;
+	readonly MEDIA_RESOURCE_TYPE_UNKNOWN: UnknownMediaResource;
+	readonly MEDIA_RESOURCE_TYPE_VIDEO: VideoMediaResource;
+}
+
+// prettier-ignore
+export type MediaResource<
+	ResourceType extends MediaResourceType = MediaResourceType,
+> = MediaResourceByType[ResourceType];
```

Comment:
Not really specific to this design proposal, but I want to call out that I learned something new about TypeScript while working on designing all of the interfaces here.

**THIS** is apparently a technique for unifying:

- Discriminated unions
- Generics
- Generics with a default which is the same discriminated union

It could help us simplify a **lot of types** (hint: most, maybe all, cases where we've defined a generic type like `Foo` and a corresponding union type like `AnyFoo`).